tests/PHPUnit/Framework/TestingEnvironment.php

```diff
@@ -43,10 +51,33 @@ public function getLogin()
     }
 }
 
+class TestingEnvironment_MakeGlobalSettingsWithFile implements EnvironmentManipulator
+{
+    private $configFileGlobal;
+    private $configFileLocal;
+    private $configFileCommon;
+
+    public function __construct($configFileGlobal, $configFileLocal, $configFileCommon)
+    {
+        $this->configFileGlobal = $configFileGlobal;
+        $this->configFileLocal = $configFileLocal;
+        $this->configFileCommon = $configFileCommon;
+    }
+
+    public function makeKernelObject($className, array $kernelObjects)
+    {
+        if ($className == 'Piwik\Application\Kernel\GlobalSettingsProvider') {
+            return new GlobalSettingsProvider($this->configFileGlobal, $this->configFileLocal, $this->configFileCommon);
+        }
+
+        return null;
+    }
+}
```

Comment:
I can put it in a new file. Boy I wish I could use anonymous classes... hey you think Piwik might require PHP 7 soon? :)